src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -603,7 +603,10 @@ private void InitializeTrainingGraph(IDataView input)
                     (string)labelType.ToString());
             }
 
-            _classCount = labelCount == 1 ? 2 : (int)labelCount;
+            if (labelCount == 1)
+                throw new InvalidOperationException("Dataset contains only 1 class. ImageClassificationTrainer requires more than 1");
+
```

Comment:
If I understand it correctly, this happens when the supplied data is bad (i.e. it has only one class of labels)
Would it be better to throw an ArgumentException?

Reply:
We throw a similar error in other places: 
https://github.com/dotnet/machinelearning/blob/1a1a62ad3d582e1ab99fe900c73b908da94af647/src/Microsoft.ML.Data/Evaluators/AucAggregator.cs#L115-L116

Producing:
`System.ArgumentOutOfRangeException: AUC is not definied 
when there is no positive class in the data`

Reply:
Yes, @harishsk , your understanding is correct about this case. I will change this asap.